### ISSUE TITLE
Unused variable '__class__' for unit test classes

### ISSUE THREAD

**Issue by atodorov** [PyCQA/pylint#1609]
### Steps to reproduce
1. 
```
import unittest


class MyTestCase(unittest.TestCase):
    def test_something(self):
        self.assertTrue(1 == 2)
```
2. 
```
pylint -rn tests/tests.py 
No config file found, using default configuration
************* Module tests
C:  1, 0: Missing module docstring (missing-docstring)
C:  4, 0: Missing class docstring (missing-docstring)
C:  5, 4: Missing method docstring (missing-docstring)
W:  4, 0: Unused variable '__class__' (unused-variable)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 9.72/10, -9.72)
```
3. 

### Current behavior
See above

### Expected behavior
No warnings reported

### pylint --version output

```
pylint --version
No config file found, using default configuration
pylint 1.8.0, 
astroid 1.5.3
Python 3.5.1 (default, Sep 15 2016, 08:30:32) 
[GCC 4.8.3 20140911 (Red Hat 4.8.3-9)]
```

installed from 7d6114fe71f1e6ab4a7c19546f380a7cb216eac9

**Comment by rogalski** [comment]
PyCQA/astroid@ced3d9de8f8bdefd4d99924fda4879b6fa511dde seems to be a likely suspect, but I cannot reproduce it using master version from both astroid and pylint.

**Comment by atodorov** [comment]
> PyCQA/astroid@ced3d9d seems to be a likely suspect, but I cannot reproduce it using master version from both astroid and pylint.

I can reproduce just fine by doing a `pip install https://github.com/PyCQA/astroid/zipball/master https://github.com/PyCQA/pylint/zipball/master` and the file above. Here's what I have locally:

```
$ python --version
Python 3.5.1

$ pip freeze
astroid==1.5.3
isort==4.2.15
lazy-object-proxy==1.2.2
mccabe==0.6.1
pluggy==0.3.1
py==1.4.31
pylint==1.8.0
six==1.10.0
tox==2.3.1
virtualenv==15.0.3
wheel==0.24.0
wrapt==1.10.8
```
